diff --git a/client.py b/client.py
index 5c10823..2b47855 100644
--- a/client.py
+++ b/client.py
@@ -122,8 +122,11 @@ class Site(object):
 				# Release candidate
 				self.version = (int(version[0]), int(version[1]), version[2])
 			elif len(version) == 3:
-				# Regular y.x.z version
-				self.version = (int(version[0]), int(version[1]), int(version[2]))
+				# patched for debian version
+				third = version[2]
+				elems = third.split('-')
+				major = int(elems[0])
+				self.version = (int(version[0]), int(version[1]), major)
 			else:
 				raise errors.MediaWikiVersionError('Unknown MediaWiki %s' % '.'.join(version))
 		else:
